e2e/various-suite/return-to-previous.spec.ts

```diff
@@ -0,0 +1,91 @@
+import { e2e } from '../utils';
+
+describe('ReturnToPrevious button', () => {
+  beforeEach(() => {
+    e2e.flows.login(Cypress.env('USERNAME'), Cypress.env('PASSWORD'));
+    cy.window().then((win) => {
+      win.localStorage.setItem('grafana.featureToggles', 'returnToPrevious=1');
+    });
+
+    cy.visit('/alerting/list?search=');
+    e2e.components.AlertRules.groupToggle().first().click();
+    e2e.components.AlertRules.toggle().click();
+    e2e.components.AlertRules.expandedContent().find('[data-testid="data-testid go to dashboard"]').click();
+    // TODO: replace data-testid in find?
+  });
+
+  it('should appear when changing context and go back to alert rule when clicking "Back"', () => {
+    // make sure the dashboard finished loading
+    cy.get('button[aria-label*="BarChart - Label Rotation & Skipping"]').should('be.visible');
+
+    // check whether all elements of RTP are available
+    e2e.components.ReturnToPrevious.buttonGroup().should('be.visible');
+    e2e.components.ReturnToPrevious.dismissButton().should('be.visible');
+    e2e.components.ReturnToPrevious.backButton().should('be.visible').click();
+
+    // go back to alert rule
+    e2e.components.AlertRules.groupToggle().first().click();
+    e2e.components.AlertRules.toggle().click();
+    e2e.components.AlertRules.expandedContent()
+      .find('[data-testid="data-testid go to dashboard"]')
+      .should('be.visible');
+    // TODO: replace data-testid in find?
```

Comment:
imo we should:
- grab the current url before we click go to dashboard with something like: `cy.url().as('rtpUrl');`
- click the return to previous button
- check the url matches the original we grabbed with: `cy.url().should('eq', '@rtpUrl');`
- check the return to previous button is no longer visible

don't think we need to do anything on the alert page itself once we've gone back

Reply:
Maybe [this](https://stackoverflow.com/questions/56790303/cy-url-and-or-cy-locationhref-does-not-return-a-string) helps. As I understand here, you should do something like:
```jsx
cy.url().as('url').then((url) => {
    // make sure the dashboard finished loading
    cy.get('button[aria-label*="BarChart - Label Rotation & Skipping"]').should('be.visible');

    // check whether all elements of RTP are available
    e2e.components.ReturnToPrevious.buttonGroup().should('be.visible');
    e2e.components.ReturnToPrevious.dismissButton().should('be.visible');
    e2e.components.ReturnToPrevious.backButton()
      .find('span')
      .contains('Back to e2e-ReturnToPrevious-test')
      .should('be.visible')
      .click();

    // check whether the RTP button leads back to alert rule
    cy.url().should('include', url);
});
```
It's failing because the URLs are not the same, but I think I'm missing the part of loading the alert rule page again 😅  Sorry for not having more knowledge / expertise on this and helping you more 😓

Reply:
@eledobleefe I had a look at your suggestion. In general this would work but the flow in `before()` already leads to the dashboard with the RTP button. So if I add your suggestion in the first test `it('should appear when changing context and go back to alert rule when clicking "Back"', () => {})` the first url (`cy.url().as('url')`) is the one for the dashboard and the second one is for the alert rule(`cy.url().should('include', url)`).  So they are not comparable. I added the code suggestion above to the test and ran it locally: 
<img width="532" alt="Screenshot 2024-03-12 at 15 57 22" src="https://github.com/grafana/grafana/assets/48948963/15867219-abdd-4564-af9f-33f6f0d09b60">